tests/moe/test_b12x_fused_moe.py

```
@@ -218,38 +245,287 @@ def test_w4a16_static_cutover_env_override_is_precision_scoped(monkeypatch):
+def test_dispatch_rejects_nvfp4_compressed_tensors_source_format():
+    from flashinfer.fused_moe.cute_dsl.blackwell_sm12x import moe_dispatch
+
+    x = torch.empty((1, 16), dtype=torch.bfloat16)
+    weight = torch.empty((1, 32, 8), dtype=torch.uint8)
+    scale = torch.empty((1, 32, 1), dtype=torch.float8_e4m3fn)
+    alpha = torch.ones((1,), dtype=torch.float32)
+    topk_ids = torch.zeros((1, 1), dtype=torch.int32)
+    topk_weights = torch.ones((1, 1), dtype=torch.float32)
+    output = torch.empty((1, 16), dtype=torch.bfloat16)
+
+    with pytest.raises(ValueError, match="compressed_tensors.*w4a16"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)